dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.hdlresolver;
+
+import java.text.MessageFormat;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is public and is useful for handle resolving,
+ * wheter a target handle identifier will be resolved into the
+ * corresponding URL (if found), otherwise will respond a null string.
+ * 
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.it)
+ *
+ */
+@RestController
+@RequestMapping(path = "/{hdlService:hdlResolver|resolve|listhandles|listprefixes}/")
+public class HdlResolverRestController {
+    static final String HDL_RESOLVER = "/hdlResolver/";
+    static final String RESOLVE = "/resolve/";
+    static final String LISTHANDLES = "/listhandles/";
+    static final String LISTPREFIXES = "/listprefixes/";
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    private HdlResolverService hdlResolverService;
+
+    /**
+     * REST GET Method used to find and retrieve the URL of a target Handle. It
+     * should return only one item, if found, else a null body value.
+     * 
+     * Generate an <code>HttpStatus.BAD_REQUEST</code> 400 Error if the handle used
+     * in path isn't valid.
+     * 
+     * The response type will be the same that is in the /xmlui/handleresolver/resolve
```

Comment:
We should remove references to XMLUI here... they don't make sense in DSpace 7 as the XMLUI doesn't exist anymore.  Instead of this reference, I'd recommend including an example request which shows how this method works & how it might respond